src/engine/QueryPlanner.cpp

```diff
@@ -329,6 +379,9 @@ std::vector<QueryPlanner::SubtreePlan> QueryPlanner::optimize(
         valuesPlan._qet->setVariableColumns(op->getVariableColumns());
         joinCandidates(std::vector{std::move(valuesPlan)});
 
+      } else if constexpr (std::is_same_v<T, GraphPatternOperation::Bind>) {
+        // the BIND operation can be handled directly
```

Comment:
// The logic of the BIND operation is implemented in the joinCandidates lambda. Reason: BIND does not add a new join operation like for the other operations above.